src/zenml/integrations/mlflow/mlflow_step_decorator.py

```diff
@@ -0,0 +1,158 @@
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import functools
+from typing import Any, Callable, Optional, Type, TypeVar, Union, cast, overload
+
+from zenml.integrations.mlflow.mlflow_utils import (
+    get_or_create_mlflow_run,
+    setup_mlflow,
+)
+from zenml.logger import get_logger
+from zenml.steps import BaseStep, StepEnvironment
+from zenml.steps.utils import STEP_INNER_FUNC_NAME
+
+logger = get_logger(__name__)
+
+
+# step entrypoint type
+F = TypeVar("F", bound=Callable[..., Any])
+
+# step class type
+S = TypeVar("S", bound=Type[BaseStep])
+
+
+@overload
+def enable_mlflow(
+    _step: S,
+) -> S:
+    """Type annotations for mlflow step decorator in case of no arguments."""
+    ...
+
+
+@overload
+def enable_mlflow(
+    *,
+    experiment_name: Optional[str] = None,
+) -> Callable[[S], S]:
+    """Type annotations for mlflow step decorator in case of arguments."""
+    ...
+
+
+def enable_mlflow(
+    _step: Optional[S] = None,
+    *,
+    experiment_name: Optional[str] = None,
+) -> Union[S, Callable[[S], S]]:
+    """Decorator to enable mlflow for a step function.
+
+    Apply this decorator to a ZenML pipeline step to enable MLflow experiment
+    tracking. The MLflow tracking configuration (tracking URI, experiment name,
+    run name) will be automatically configured before the step code is executed,
+    so the step can simply use the `mlflow` module to log metrics and artifacts,
+    like so:
+
+    .. highlight:: python
+    .. code-block:: python
+
+        @enable_mlflow
+        @step
+        def tf_evaluator(
+            x_test: np.ndarray,
+            y_test: np.ndarray,
+            model: tf.keras.Model,
+        ) -> float:
+            _, test_acc = model.evaluate(x_test, y_test, verbose=2)
+            mlflow.log_metric("val_accuracy", test_acc)
+            return test_acc
+
+    All MLflow artifacts and metrics logged from all the steps in a pipeline
+    run are by default grouped under a single experiment named after the
+    pipeline. To log MLflow artifacts and metrics from a step in a separate
+    MLflow experiment, pass a custom `experiment_name` argument value to the
+    decorator.
+
+    Args:
+        _step: The decorated step class.
+        experiment_name: optional mlflow experiment name to use for the step.
+            If not provided, the name of the pipeline in the context of which
+            the step is executed will be used as experiment name.
+
+    Returns:
+        The inner decorator which enhaces the input step class with mlflow
+        tracking functionality
+    """
+
+    def inner_decorator(_step: S) -> S:
+
+        logger.debug(
+            f"Applying 'enable_mlflow' decorator to step {_step.__name__}"
```

Comment:
```suggestion
            "Applying 'enable_mlflow' decorator to step %s", _step.__name__"
```

Reply:
I know this is probably not really that relevant for us, but I saw you used both approaches in this Pr and thought I'd uniformize it (if thats a word)